WKTR installation on iOS devices is not working

pvollan opened the pull request:
#### 8d65dc7128464652cb754c957a64c898221d71bf
<pre>
WKTR installation on iOS devices is not working
<a href="https://bugs.webkit.org/show_bug.cgi?id=276161">https://bugs.webkit.org/show_bug.cgi?id=276161</a>
<a href="https://rdar.apple.com/131015054">rdar://131015054</a>

Reviewed by David Kilzer.

Due to a code signing issue, WKTR installation on iOS devices is not working after &lt;<a href="https://commits.webkit.org/280387@main">https://commits.webkit.org/280387@main</a>&gt;.
The changes in 280387@main were only intended for iOS simulator, and this patch addresses that.

* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS-simulator.entitlements: Renamed from Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-visionOS.entitlements.
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS.entitlements:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/280636@main">https://commits.webkit.org/280636@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/390548e7c2a7569e8998e77aca2272197f074f69

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57102 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36430 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9577 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60722 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7545 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/59230 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44054 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7735 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46236 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5304 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59132 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34200 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/49289 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27097 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30981 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6619 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6550 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52937 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6889 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62403 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1015 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6991 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53495 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1019 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49342 "Build is in progress. Recent messages:OS: Ventura (13.6.7), Xcode: 14.3; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-layout-tests-in-stress-mode") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53549 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/857 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8528 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32259 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33344 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34429 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33090 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
```diff
@@ -38,16 +38,18 @@ STRIP_STYLE = debugging;
+CODE_SIGN_ENTITLEMENTS[sdk=iphonesimulator*] = Configurations/WebKitTestRunnerApp-iOS-simulator.entitlements;
+CODE_SIGN_ENTITLEMENTS[sdk=iphoneos*] = Configurations/WebKitTestRunnerApp-iOS.entitlements;
 CODE_SIGN_ENTITLEMENTS[sdk=appletv*] = Configurations/WebKitTestRunnerApp-watchOS.entitlements;
 CODE_SIGN_ENTITLEMENTS[sdk=watch*] = Configurations/WebKitTestRunnerApp-watchOS.entitlements;
-CODE_SIGN_ENTITLEMENTS[sdk=xr*] = Configurations/WebKitTestRunnerApp-visionOS.entitlements;
+CODE_SIGN_ENTITLEMENTS[sdk=xr*] = Configurations/WebKitTestRunnerApp-iOS.entitlements;
```
Are you sure this change is correct?

The `Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-visionOS.entitlements` file still exists.

ddkilzer commented on Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
```diff
@@ -474,10 +474,10 @@
 		E1BA671D1742DA5A00C20251 /* Carbon.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Carbon.framework; path = System/Library/Frameworks/Carbon.framework; sourceTree = SDKROOT; };
 		E1C642C417CBCD4C00D66A3C /* WebKitTestRunnerPasteboard.mm */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.objcpp; path = WebKitTestRunnerPasteboard.mm; sourceTree = "<group>"; };
 		E1C642C517CBCD4C00D66A3C /* WebKitTestRunnerPasteboard.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = WebKitTestRunnerPasteboard.h; sourceTree = "<group>"; };
-		E3660D6A2C2F663800BA7138 /* WebKitTestRunnerApp-visionOS.entitlements */ = {isa = PBXFileReference; lastKnownFileType = text.plist.entitlements; path = "WebKitTestRunnerApp-visionOS.entitlements"; sourceTree = "<group>"; };
```
Ditto.  Are you sure that you want to remove `WebKitTestRunnerApp-visionOS.entitlements` from the project?

ddkilzer commented on Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
```diff
@@ -942,8 +942,8 @@
 				A18510381B9ADF2200744AEB /* WebKitTestRunner.xcconfig */,
+				E3C2C0312C34DEFC000C7F2E /* WebKitTestRunnerApp-iOS-simulator.entitlements */,
 				9B0D132E2036D346008FC8FB /* WebKitTestRunnerApp-iOS.entitlements */,
-				E3660D6A2C2F663800BA7138 /* WebKitTestRunnerApp-visionOS.entitlements */,
```
Ditto.  Are you sure that you want to remove `WebKitTestRunnerApp-visionOS.entitlements` from the project?

pvollan replied:
Yes, it has been renamed to WebKitTestRunnerApp-iOS-simulator.entitlements. The intention is to let visionOS use WebKitTestRunnerApp-iOS.entitlements for now.

Thanks for reviewing!

ddkilzer reviewed (commented):
I don't think you intended to remove `WebKitTestRunnerApp-visionOS.entitlements`, did you?

If you did, you should also delete the file on disk at `Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-visionOS.entitlements`.

Otherwise, everything else looks okay.

ddkilzer reviewed (approved):
r=me

pvollan committed:
WKTR installation on iOS devices is not working
https://bugs.webkit.org/show_bug.cgi?id=276161
rdar://131015054

Reviewed by David Kilzer.

Due to a code signing issue, WKTR installation on iOS devices is not working after <https://commits.webkit.org/280387@main>.
The changes in 280387@main were only intended for iOS simulator, and this patch addresses that.

* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS-simulator.entitlements: Renamed from Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-visionOS.entitlements.
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS.entitlements:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/280636@main
